File: 201~300/242.ValidAnagram/step3.cpp
class Solution {
public:
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) {
      return false;
    }
    vector<int> letter_to_frequency(26);
    for (int i = 0; i < s.size(); i++) {
      letter_to_frequency[s[i] - 'a']++;
      letter_to_frequency[t[i] - 'a']--;
    }

    for (int num : letter_to_frequency) {
      if (num != 0) {
        return false;
      }
    }
    return true;
  }
};

File: 301~400/347.TopKFrequentElements/step3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_to_freq = {}
        for num in nums:
            frequency = num_to_freq.get(num, 0)
            num_to_freq[num] = frequency + 1
        return sorted(num_to_freq, key=lambda x:num_to_freq[x], reverse=True)[:k]